src/Repository/ProductCommentCriterionRepository.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/afl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\ProductComment\Repository;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\Module\ProductComment\Entity\ProductCommentCriterion;
+use PrestaShop\PrestaShop\Adapter\Presenter\Product\ProductLazyArray;
+use Product;
+use Validate;
+use Tools;
+
+class ProductCommentCriterionRepository
+{
+    /**
+     * @var Connection the Database connection
+     */
+    private $connection;
+
+    /**
+     * @var string the Database prefix
+     */
+    private $databasePrefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $databasePrefix
+     */
+    public function __construct(Connection $connection, $databasePrefix)
+    {
+        $this->connection = $connection;
+        $this->databasePrefix = $databasePrefix;
+    }
+
+    /**
+     * @param int|Product|ProductLazyArray $product
+     * @param int $idLang
+     *
+     * @return array
+     *
+     * @throws \PrestaShopException
+     */
+    public function getByProduct($product, $idLang)
+    {
+        $idProduct = is_object($product) ? $product->id : (int) $product;
+
+        if (!Validate::isUnsignedId($idProduct) ||
+            !Validate::isUnsignedId($idLang)) {
+            throw new \Exception(Tools::displayError());
```

Comment:
Use namespace ?
Tools::displayError() throw PrestaShopException when _PS_MODE_DEV_ is enabled but instead should throw an InvalidArgumentException here as the problem come from bad argument ?

Reply:
Oh thanks, actually this piece of code is from old code and should have been removed, doing it right now